### PR TITLE
Fix various typing issues

### PR DESCRIPTION
**Pull request by ethanshar**
## Description
Fix various typing issues

## Changelog
Fix Button iconSource type, Card.Section ImageProps to not require source and Text to support animated text style

### CODE REVIEW

File: src/components/text/index.tsx [hunk 1]
```diff
@@ -39,6 +39,7 @@ export type TextProps = RNTextProps & TypographyModifiers & ColorsModifiers & Ma
    */
   animated?: boolean;
   textAlign?: string;
+  style?: StyleProp<TextStyle | Animated.AnimatedProps<TextStyle>>;
```
Comment (M-i-k-e-l) [inline src/components/text/index.tsx:42]: You need to `Omit<RNTextProps, 'style'>` in `TextProps`

File: src/components/card/CardSection.tsx [hunk 2]
```diff
@@ -44,7 +44,7 @@ export type CardSectionProps = ViewProps & {
   /**
    * Other image props that will be passed to the image
    */
-  imageProps?: ImageProps;
+  imageProps?: Partial<ImageProps>;
```
Comment (M-i-k-e-l) [inline src/components/card/CardSection.tsx:47]: This: `imageProps?: Omit<ImageProps, 'source' | 'style'>;` feels a bit more accurate, WDYT?

File: src/components/text/index.tsx (hunk 1 above)
Comment (ethanshar) [reply src/components/text/index.tsx:41]: Yes, weirdly it didn't throw an error on that, anyway fixed it.

Comment (M-i-k-e-l) [reply src/components/text/index.tsx:41]: We should move more screens to tsx (found it there)

File: src/components/card/CardSection.tsx (hunk 2 above)
Comment (M-i-k-e-l) [reply src/components/card/CardSection.tsx:47]: I know, but I'm not sure it's intentional, approving

**Pull request merged**